market_nextdoor_api/views.py

```diff
@@ -341,3 +344,35 @@ def update_preorder(preorder, data):
     preorder_data.save()
     return Response(preorder_data.data)
   return Response(status=status.HTTP_400_BAD_REQUEST)
+
+#USDA Market locations apikey=rTS5CqcxKA
+def get_market_locations(request, zipcode):
+  USDA_API_KEY = os.environ.get('USDA_API_KEY')
+  BASE_URL = "https://www.usdalocalfoodportal.com/api/farmersmarket/"
+  request_url = f'{BASE_URL}?apikey={USDA_API_KEY}&zip={zipcode}&radius=50'
+  head = {
+        'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.85 Safari/537.36'
+    }
+  try:
+    response = requests.get(request_url, headers=head)
+    response.raise_for_status()
+    data = response.json()
+    #Neat list comprehension - this parses the data coming in from the API, the syntax is a bit
+    #different in that the iteration is after declaring the instance of a library on line 374
+    refined_data = [
+            {
+                'market_name': item.get('listing_name', ''),
+                'address': item.get('location_address', ''),
+                'lat': item.get('location_x', ''),
+                'lon': item.get('location_y', ''),
+                'website': item.get('media_website', ''),
+                'zipcode': item.get('location_zipcode', ''),
+                'phone': item.get('contact_phone', '')
+            }
+            for item in data.get('data', [])
+        ]
```

Comment:
Tomorrow we can look at refactoring this and possibly using the Serializer file to use SRP. Great work!